scripts/temp_pub.py

```diff
@@ -12,17 +12,28 @@ def internal_temperature_publisher():
     msg_temp = Temperature()
     msg_temp.header.frame_id = "internal_temperature_sensor_link"
     rate = rospy.Rate(10)
+    prev_temp = 0
+
     while not rospy.is_shutdown():
         try:
             msg_temp.header.stamp = rospy.Time.now()
             sensor_data = temper.get_temperature()
-            msg_temp.temperature = float(sensor_data[0])
+
+            if sensor_data is not None and sensor_data[0] is not None:
+                msg_temp.temperature = float(sensor_data[0])
```

Comment:
is the dynamic casting required? What is the data type of the incoming data?

Reply:
yes, the incoming data type for temperature is str

Reply:
the dynamic casting should happen only once then

Reply:
is this fixed?

Reply:
yes, didn't sync changes previously. changes are synced now